ruff.toml

```diff
@@ -6,8 +6,25 @@ extend-select = [
 	"C901",
 	"PERF401",
 	"W",
+ 
+ 	# Ensure modern type annotation syntax and best practices
+	# Not including those covered by type-checkers or exclusive to Python 3.11+
+	"FA", # flake8-future-annotations
+	"F404", # late-future-import
+	"PYI", # flake8-pyi
+	"UP006", # non-pep585-annotation
+	"UP007", # non-pep604-annotation
+	"UP010", # unnecessary-future-import
+	"UP035", # deprecated-import
+	"UP037", # quoted-annotation
+	"UP043", # unnecessary-default-type-args
+
+	# local
```

Comment:
I appreciate adding the marker to help avoid merge conflicts.

FYI - I've historically paired `# upstream` indicators with `# local`. I've always struggled with the naming of `# upstream` and considered using `# skeleton` or other alternatives, including no indicator. Here, we're adopting `# local` without `# upstream`, which may be fine, but now it's somewhat inconsistent with the pattern in other files.

In light of that, what do you think is the best approach for skeleton more generally?

Reply:
You're right, having both markers is probably best.

I've definitely been very confused as a new contributor about what "upstream" meant, especially in `pytest.ini`.
I first thought it was about working around upstream issues.

`# skeleton` sounds fine to me, but for this PR I'll be consistent with what's already there.